ql/src/java/org/apache/hadoop/hive/ql/optimizer/StatsOptimizer.java

```diff
@@ -397,6 +397,11 @@ else if (getGbyKeyType(cgbyOp) == GbyKeyType.CONSTANT && rowCnt == 0) {
           if (udaf instanceof GenericUDAFSum) {
             // long/double/decimal
             ExprNodeDesc desc = aggr.getParameters().get(0);
+            // return null for SUM(1), when the table is empty. Without this, category = LONG, and the result is 0
+            // instead of NULL.
+            if (desc instanceof ExprNodeConstantDesc && rowCnt == 0) {
+              return oneRow.add(null);
```

Comment:
this `return` doesn't look right to me - it may work correctly if you around only selecting the aggregate - but what happens if you have more?
I believe you should execute `oneRow.add(null)` ; bail out the rest of this conditional - and let the code continue with the remaining aggregates if there is any